vp9hdec: fix crash accessing reference frame's CmSurface2D

wlin70 opened the pull request:
Top decoder may call vaDestroy/CreateContext to recreate media ctx and MdfDevice
if target frame size becomes larger.
CmSurface2D of each reference frame will be reclaimed sequentially and segment fault
accessing them.

Fix by using TSC and CmDevice pointer to detect context change and reallocate these CmSurface2D.

Signed-off-by: Wei Lin wei.w.lin@intel.com

chivakker commented on src/vp9hdec/decode_hybrid_vp9.cpp:
```diff
@@ -2247,6 +2250,37 @@ VAStatus Intel_HybridVp9Decode_MdfHost_Initialize (
+    struct object_surface                   *pRefSurface,
+    INTEL_DECODE_HYBRID_VP9_MDF_FRAME_SOURCE *pFrameSource)
+{
+    VAStatus                                     eStatus = VA_STATUS_SUCCESS;
```
remove the extra spaces

chivakker commented on src/vp9hdec/decode_hybrid_vp9.cpp:
```diff
@@ -2350,6 +2394,19 @@ VAStatus Intel_HybridVp9Decode_MdfHost_PadFrame (
+    if ((pFrameResource->pMdfDevice != pMdfDecodeEngine->pMdfDevice) ||
+        (pFrameResource->iMdfDeviceTsc != pMdfDecodeEngine->iMdfDeviceTsc))
+    {
+	// Decoder like libyami may call vaContextDestroy/Create to recreate media ctx and MdfDevice
```
remove libyami reference from comment.  This code can be executed by any player that does a port reconfiguration due to new size of surfaces.

chivakker commented:
please remove references to libyami in the commit message and comments in the code.  The functionality is available to any application that does port reconfiguration.

wlin70 commented:
Changed following Daniel's comments.

wlin70 committed:
vp9hdec: fix crash accessing reference frame's CmSurface2D

Top decoder may call vaDestroy/CreateContext to recreate media ctx and MdfDevice
if target frame size becomes larger.
CmSurface2D of each reference frame will be reclaimed sequentially and segment fault
accessing them.

Fix by using TSC and CmDevice pointer to detect context change and reallocate these CmSurface2D.

Signed-off-by: Wei Lin <wei.w.lin@intel.com>

chivakker commented:
lgtm +1

seanvk commented:
lgtm +2
